src/lib/parser.c

```diff
@@ -1206,14 +1354,225 @@ RdbStatus elementQuickList(RdbParser *p) {
     }
 }
 
-RdbStatus elementZiplist(RdbParser *p) {
+RdbStatus elementListZL(RdbParser *p) {
+    BulkInfo *ziplistBulk;
+
+    IF_NOT_OK_RETURN(rdbLoadString(p, RQ_ALLOC_APP_BULK, NULL, &ziplistBulk));
+
+    /*** ENTER SAFE STATE ***/
+
+    if (RDB_STATUS_ERROR == listZiplistItem(p, ziplistBulk))
+        return RDB_STATUS_ERROR;
+
+    return nextParsingElement(p, PE_END_KEY);
+}
+
+RdbStatus elementHash(RdbParser *p) {
```

Comment:
these functions handle the whole key (of a certain encoding), not just a single element.
i.e. they're alliteratively called in steps to consume just a portion each time, but the first portion (the header) is not an element.
consider renaming..

Reply:
Not sure i understand. The term `element` is mainly aligned with RDB data-structures and opcodes. In this case elementHash will be called on opcode `RDB_TYPE_HASH` and will know how to parse the hash from start to the end, including its header. It will be done in steps, and yet.

Reply:
in this case i think RDB_TYPE_HASH is a "key" not an "element".
i think we refer to the sub-key "fields" as "elements"
so what i meant is that since this function handles a key (not a single field), it should be renamed.
maybe `handleHashKey` or `handleHashType`?

Reply:
I tried to have common naming convention across the "parsing-elements" (see peInfo[] table, at parser.c and in readme.md). That is, the term "element" is heavily used. Note that some of the "elements" in the RDB doesn't represent keys. Maybe we can abbreviate the names to `peHash`, `peQuickList`, etc.

Reply:
so "element" is a generic term for some unit in the rdb, and it can be a key, a field, or anything else (aux hint)? 
it sounds a bit odd to me, but it could be just me.
in which case, many places where i was thinking of "elements" they should be "fields" (in hash) or "members" (in set).
i'm not sure.. i'll leave this up to you.

Reply:
Actually I borrowed the "element" term from XML terminology. I didn't want to reuse the term `handler` as it is already heavily used by the application handlers. Though, I see your point. Please let's take it on a distinct PR.